catalyst-gateway/bin/src/service/common/objects/fragment_status.rs

```diff
@@ -0,0 +1,50 @@
+use poem_openapi::{types::Example, Object, Union};
+
+use crate::service::common::objects::{block::BlockDate, hash::Hash};
+
+#[derive(Object)]
+#[oai(example = false)]
+/// Fragment is pending.
+pub(crate) struct StatusPending;
+
+#[derive(Object)]
+#[oai(example = true)]
+/// Fragment was rejected.
+pub(crate) struct StatusRejected {
+    /// Reason the fragment was rejected.
+    pub reason: String,
+}
+
+impl Example for StatusRejected {
+    fn example() -> Self {
+        Self {
+            reason: "Transaction malformed".to_string(),
+        }
+    }
+}
+
+#[derive(Object)]
+#[oai(example = true)]
+/// Fragment is included in a block.
+pub(crate) struct StatusInABlock {
+    pub date: BlockDate,
+    pub hash: Hash,
```

Comment:
Are you sure that `StatusInABlock` should have the following structure ?
Seems it will have the following JSON format:
```json
{
  "date": <date>,
  "hash": 
   {
    "hash": <hash_value>  
   }
}
```

Reply:
I took it from: https://github.com/input-output-hk/catalyst-core/blob/main/src/jormungandr/jormungandr-lib/src/interfaces/fragment_log.rs#L33

It's actually:
```json
{
  "date": <date>,
  "block": {
    "hash": <hash>
  }
}
```